packages/integrations/vue/src/index.ts

```diff
@@ -39,71 +33,55 @@ function getJsxRenderer(): AstroRenderer {
 	};
 }
 
-function virtualAppEntrypoint(options: ViteOptions) {
+function virtualAppEntrypoint(options?: Options): Plugin {
 	const virtualModuleId = 'virtual:@astrojs/vue/app';
 	const resolvedVirtualModuleId = '\0' + virtualModuleId;
-	let getExports: (id: string) => Promise<string[]>;
+
+	let isBuild: boolean;
+	let root: string;
+
 	return {
 		name: '@astrojs/vue/virtual-app',
-		buildStart() {
-			if (!getExports) {
-				getExports = async (id: string) => {
-					const info = await this.load.call(this, { id });
-					return info.exports ?? [];
-				};
-			}
+		config(_, { command }) {
+			isBuild = command === 'build';
 		},
-		configureServer(server) {
-			if (!getExports) {
-				getExports = async (id: string) => {
-					const mod = await server.ssrLoadModule(id);
-					return Object.keys(mod) ?? [];
-				};
-			}
+		configResolved(config) {
+			root = config.root;
 		},
 		resolveId(id: string) {
 			if (id == virtualModuleId) {
 				return resolvedVirtualModuleId;
 			}
 		},
-		async load(id: string) {
-			const noop = `export const setup = (app) => app;`;
+		load(id: string) {
 			if (id === resolvedVirtualModuleId) {
-				if (options.appEntrypoint) {
-					try {
-						let resolved;
-						if (options.appEntrypoint.startsWith('.')) {
-							resolved = await this.resolve(
-								fileURLToPath(new URL(options.appEntrypoint, options.root))
-							);
-						} else {
-							resolved = await this.resolve(options.appEntrypoint, fileURLToPath(options.root));
-						}
-						if (!resolved) {
-							// This error is handled below, the message isn't shown to the user
-							throw new Error('Unable to resolve appEntrypoint');
-						}
-						const exports = await getExports(resolved.id);
-						if (!exports.includes('default')) {
-							options.logger.warn(
-								`appEntrypoint \`${options.appEntrypoint}\` does not export a default function. Check out https://docs.astro.build/en/guides/integrations-guide/vue/#appentrypoint.`
-							);
-							return noop;
-						}
-						return `export { default as setup } from "${resolved.id}";`;
-					} catch {
-						options.logger.warn(
-							`Unable to resolve appEntrypoint \`${options.appEntrypoint}\`. Does the file exist?`
-						);
-					}
+				if (options?.appEntrypoint) {
+					const appEntrypoint = options.appEntrypoint.startsWith('.')
+						? path.resolve(root, options.appEntrypoint)
+						: options.appEntrypoint;
+
+					return `\
```

Comment:
Accidental escape character?

```suggestion
					return `
```

Reply:
The escape is mainly so the generated code is a little cleaner 😅 It's used in template literals ending char to signal "no new line", so we can avoid an initial new line.

Reply:
TIL!